src/yarp.c

```diff
@@ -12773,8 +12773,8 @@ parse_expression(yp_parser_t *parser, yp_binding_power_t binding_power, const ch
 }
 
 static yp_node_t *
-parse_program(yp_parser_t *parser) {
-    yp_parser_scope_push(parser, true);
+parse_program(yp_parser_t *parser, bool eval) {
+    yp_parser_scope_push(parser, !eval);
```

Comment:
Instead of passing this in and passing it all of the way through all the `yp_parse` calls, I would prefer if this derived it from the number of scopes currently active. That way we don't have to change the signature everywhere.